src/ComputeSharp.D2D1/Shaders/Interop/D2D1ResourceTextureManager.cs

```diff
@@ -0,0 +1,487 @@
+    /// <param name="resourceTextureManager">A pointer to the <c>ID2D1ResourceTextureManager</c> instance to use.</param>
+    /// <param name="resourceId">The resource id for the resource to create.</param>
+    /// <param name="extents">The extents of the resource to create.</param>
+    /// <param name="bufferPrecision">The buffer precision for the resource to create.</param>
+    /// <param name="channelDepth">The channel depth for the resource to create.</param>
+    /// <param name="filter">The filter for the resource to create.</param>
+    /// <param name="extendModes">The extend modes for the resource to create.</param>
+    /// <param name="data">The data to load in the resource to create.</param>
+    /// <param name="strides">The strides for the data supplied for the resource to create.</param>
+    /// <remarks>
+    /// Depending on internal state, <paramref name="resourceTextureManager"/> might immediately create the resource
+    /// texture, or it might buffer the data internally and then create the resource texture at a later time.
+    /// </remarks>
+    public static void Initialize(
+        void* resourceTextureManager,
+        Guid resourceId,
+        ReadOnlySpan<uint> extents,
+        D2D1BufferPrecision bufferPrecision,
+        D2D1ChannelDepth channelDepth,
+        D2D1Filter filter,
+        ReadOnlySpan<D2D1ExtendMode> extendModes,
+        ReadOnlySpan<byte> data,
+        ReadOnlySpan<uint> strides)
+    {
+        // Validate parameters early from here too to avoid AVs in the native method
+        if (extents.Length == 0 ||
+            extendModes.Length == 0 ||
+            extents.Length != extendModes.Length ||
+            (data.Length > 0 && strides.Length != extents.Length - 1))
+        {
+            Marshal.ThrowExceptionForHR(E.E_INVALIDARG);
+        }
+
+        fixed (uint* pExtents = extents)
+        fixed (D2D1ExtendMode* pExtendModes = extendModes)
+        fixed (byte* pData = data)
+        fixed (uint* pStrides = strides)
+        {
+            D2D1_RESOURCE_TEXTURE_PROPERTIES d2D1ResourceTextureProperties;
+            d2D1ResourceTextureProperties.extents = pExtents;
+            d2D1ResourceTextureProperties.dimensions = (uint)extents.Length;
+            d2D1ResourceTextureProperties.bufferPrecision = (D2D1_BUFFER_PRECISION)bufferPrecision;
+            d2D1ResourceTextureProperties.channelDepth = (D2D1_CHANNEL_DEPTH)channelDepth;
+            d2D1ResourceTextureProperties.filter = (D2D1_FILTER)filter;
+            d2D1ResourceTextureProperties.extendModes = (D2D1_EXTEND_MODE*)pExtendModes;
+
+            int hresult = ((ID2D1ResourceTextureManager*)resourceTextureManager)->Initialize(
+                resourceId: &resourceId,
+                resourceTextureProperties: &d2D1ResourceTextureProperties,
+                data: pData,
+                strides: pStrides,
+                dataSize: (uint)data.Length);
+
+            Marshal.ThrowExceptionForHR(hresult);
```

Comment:
does this method also need a `GC.KeepAlive(this)` ?

Reply:
No, this and the one below are static, so there's no `this` to keep alive 😄